src/codegen/intrin.cc

```diff
@@ -0,0 +1,74 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file intrin.cc
+ * \brief Support for intrinsic registry.
+ */
+#include <tvm/base.h>
+#include <tvm/intrin.h>
+
+#include <memory>
+#include <atomic>
+#include <mutex>
+#include <unordered_set>
+
+namespace dmlc {
+// enable registry
+DMLC_REGISTRY_ENABLE(tvm::Intrin);
+}  // namespace dmlc
+
+namespace tvm {
+
+// single manager of intrinsic information.
+struct IntrinManager {
+  // mutex to avoid registration from multiple threads.
+  // recursive is needed for trigger(which calls UpdateAttrMap)
+  std::recursive_mutex mutex;
+  // global intrinsic counter
+  std::atomic<int> op_counter{0};
+  // storage of additional attribute table.
+  std::unordered_map<std::string, std::unique_ptr<dmlc::any> > attr;
+  // get singleton of the
+  static IntrinManager* Global() {
+    static IntrinManager inst;
```

Comment:
I see, but it will have a memory leak

Reply:
yes, but IMHO, since it's a static, an allocation happens only once at the first call for this function, memory is valid until the end of the process and finally, all the memory pages returned to the free space by the kernel, when process is finished. In other case, when we use object with static storage duration and without trivial destructor "we are exposed to the risk of accessing objects after the end of their lifetime. "
"https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables
Thanks.

Reply:
Thanks, make sense to me